### PR TITLE
SPUI-1453 > Disclaimer

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/shared/helpers/html_tag_parser.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Map<String, String> htmlTagParser(String textToParse) {
```
Comment (VovaMedvediev) [inline lib/shared/helpers/html_tag_parser.dart:1]: actually, it's not html tag parser. It's our own parser. So name it something like disclaimerQuestionsParser

File: lib/service/services/disclaimer/service/services/disclamer_service.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      logger.log(contract, message);
+      rethrow;
+    }
+  } catch (e) {
+    logger.log(transport, message);
+    rethrow;
+  }
+}
+
+Future<void> postDisclaimersService(
```
Comment (VovaMedvediev) [inline lib/service/services/disclaimer/service/services/disclamer_service.dart:35]: extract to separate service please

File: lib/app/shared/features/disclaimer/view/disclaimer.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  required BuildContext context,
+  required String primaryText,
+  required String secondaryText,
+  required String primaryButtonName,
+}) {
+  final state = context.read(disclaimerNotipod);
+
+  sShowDisclaimerPopup(
+    context,
+    image: imageAsset != null
```
Comment (VovaMedvediev) [inline lib/app/shared/features/disclaimer/view/disclaimer.dart:25]: use SizedBox(
             height: 80,
             width: 80,
             child: imageAsset != null ? Image.network(
               imageAsset,
             ) : Image.asset(
               disclaimerAsset,
             ),
           )
instead

File: lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+import '../view/disclaimer.dart';
+import 'disclaimer_notipod.dart';
+import 'disclaimer_state.dart';
+
+class DisclaimerNotifier extends StateNotifier<DisclaimerState> {
+  DisclaimerNotifier({
+    required this.read,
+  }) : super(
+          const DisclaimerState(
+            disclaimers: <DisclaimerModel>[],
```
Comment (VovaMedvediev) [inline lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart:25]: use @Default() instead

File: lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+          title: disclaimers[0].title,
+          imageUrl: disclaimers[0].imageUrl,
+          questions: disclaimers[0].questions,
+          activeButton: _checkActiveButtonStatus(disclaimers[0].questions),
+        );
+      }
+
+      if (state.disclaimers != null) {
+        if (state.disclaimers!.isNotEmpty) {
+          Timer(Duration.zero, () {
```
Comment (VovaMedvediev) [inline lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart:72]: For what reason do you need Timer here?

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/shared/helpers/html_tag_parser.dart (hunk 1 above)
Comment (DenisMartych) [reply lib/shared/helpers/html_tag_parser.dart:1]: done, rename file as disclaimer_questions_parser and move to disclaimers -> helpers

File: lib/service/services/disclaimer/service/services/disclamer_service.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/service/services/disclaimer/service/services/disclamer_service.dart:35]: done

File: lib/app/shared/features/disclaimer/view/disclaimer.dart (hunk 3 above)
Comment (DenisMartych) [reply lib/app/shared/features/disclaimer/view/disclaimer.dart:25]: done

File: lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart (hunk 4 above)
Comment (DenisMartych) [reply lib/app/shared/features/disclaimer/notifier/disclaimer_notifier.dart]: done

**Review by VovaMedvediev — APPROVED** [review]

**Pull request merged**